Ignore warnings from eigen in umat example

### Motivation

See the failure here https://civet.inl.gov/job/3538333/:

```
solid_mechanics/examples/umat_build.build: Working directory: /opt/civet1/build/moose/modules/solid_mechanics/examples/umat_build
solid_mechanics/examples/umat_build.build: Running command: make
solid_mechanics/examples/umat_build.build: Environment variable(s): OMP_NUM_THREADS="1" OMPI_MCA_orte_tmpdir_base="/tmp/tester_3n3wge_7" PRTE_MCA_rmaps_default_mapping_policy=":oversubscribe"
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: ################################################################################
solid_mechanics/examples/umat_build.build: Begin runner_run output
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: Compiling C++ (in opt mode) umat.C...
solid_mechanics/examples/umat_build.build: In file included from umat.C:13:
solid_mechanics/examples/umat_build.build: In file included from include/build_stiffness_matrix.h:12:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/Eigen:2:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/Sparse:26:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/SparseCore:61:
solid_mechanics/examples/umat_build.build: /opt/civet1/build/libmesh/installed/include/Eigen/src/SparseCore/TriangularSolver.h:273:13: error: variable 'count' set but not used [-Werror,-Wunused-but-set-variable]
solid_mechanics/examples/umat_build.build:   273 |       Index count = 0;
solid_mechanics/examples/umat_build.build:       |             ^
solid_mechanics/examples/umat_build.build: In file included from umat.C:13:
solid_mechanics/examples/umat_build.build: In file included from include/build_stiffness_matrix.h:12:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/Eigen:2:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/Sparse:29:
solid_mechanics/examples/umat_build.build: In file included from /opt/civet1/build/libmesh/installed/include/Eigen/SparseLU:35:
solid_mechanics/examples/umat_build.build: /opt/civet1/build/libmesh/installed/include/Eigen/src/SparseLU/SparseLU_heap_relax_snode.h:78:9: error: variable 'nsuper_et_post' set but not used [-Werror,-Wunused-but-set-variable]
solid_mechanics/examples/umat_build.build:    78 |   Index nsuper_et_post = 0; // Number of relaxed snodes in postordered etree 
solid_mechanics/examples/umat_build.build:       |         ^
solid_mechanics/examples/umat_build.build: /opt/civet1/build/libmesh/installed/include/Eigen/src/SparseLU/SparseLU_heap_relax_snode.h:79:9: error: variable 'nsuper_et' set but not used [-Werror,-Wunused-but-set-variable]
solid_mechanics/examples/umat_build.build:    79 |   Index nsuper_et = 0; // Number of relaxed snodes in the original etree 
solid_mechanics/examples/umat_build.build:       |         ^
solid_mechanics/examples/umat_build.build: 3 errors generated.
solid_mechanics/examples/umat_build.build: make: *** [/opt/civet1/build/moose/framework/build.mk:167: umat.x86_64-apple-darwin13.4.0.opt.lo] Error 1
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: ################################################################################
solid_mechanics/examples/umat_build.build: End runner_run output
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: ################################################################################
solid_mechanics/examples/umat_build.build: Begin tester output
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: Exit Code: 2
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: Tester failed, reason: EXIT CODE 2 != 0
solid_mechanics/examples/umat_build.build: 
solid_mechanics/examples/umat_build.build: ################################################################################
solid_mechanics/examples/umat_build.build: End tester output
solid_mechanics/examples/umat_build.build: 
[4.295s]     FAIL solid_mechanics/examples/umat_build.build FAILED (EXIT CODE 2 != 0) [solid_mechanics-opt]
```

If building JIT with `-Werror` on new clang, we get warnings from eigen that become errors.

### Design

Ignore warnings from eigen in the umat example.

### Impact

Passing tests with `-Werror` as we're not going to fix eigen.

--- a/modules/solid_mechanics/examples/umat_build/include/build_stiffness_matrix.h
+++ b/modules/solid_mechanics/examples/umat_build/include/build_stiffness_matrix.h
@@ -9,7 +9,9 @@
 
 #pragma once
 
+#include "libmesh/ignore_warnings.h"
 #include <Eigen/Eigen>
+#include "libmesh/restore_warnings.h"
 
 /**
  * build a 6x6 representation of the stiffness tensor in C from the

